### ISSUE TITLE
Why does the loss function keep increasing when I am training with Adam optimizer (using RMSE)

### ISSUE THREAD

**Issue by glotm** [KindXiaoming/pykan#449]
```python
class KAN_NET(nn.Module):
    def __init__(self,width,device="cpu"):
        super(KAN_NET, self).__init__()
        self.kan_layer=MultKAN(width=width,device=device)
        
        
    def forward(self,x):
        return self.kan_layer(x)
```
```python
device=torch.device(use_device)


for i in range(1,length+1):
        featrue_file_path=featrue_file_folder_path+"/"+station_name+"/"+station_name+"_"+str(i)+".csv"
        df=pd.read_csv(featrue_file_path).iloc[:,1:]
        scaler=MinMaxScaler()
        scaler.fit(df.iloc[:,1:-1])
        df.iloc[:,1:-1]=scaler.transform(df.iloc[:,1:-1])
        # print(df)

        train_features, train_labels, test_features, test_lebels=divide_data(df,split_time)
        train_features=train_features.to(device)
        train_labels=train_labels.to(device)
        test_features=test_features.to(device)
        test_lebels=test_lebels.to(device)

        net=KAN_NET([train_features.shape[1],16,16,1],device=device)

        criteon=nn.MSELoss().to(device)
        optimizer=torch.optim.Adam(net.parameters(),lr)
        
        
        # print(net(train_features[1:5,:]))


        cnt=0
        with tqdm(range(epochs),desc=f"Epoch") as tbar:
            for epoch in tbar:
                for j in range(0,train_features.shape[0],banch_size):
                    k=min(train_features.shape[0],j+banch_size-1)

                    optimizer.zero_grad()
                    output=net(train_features[j:k+1,:])
                    output=output.squeeze()
                    # print(output.shape,train_labels[j:k+1].shape)
                    train_loss=criteon(output,train_labels[j:k+1])
                    train_loss.backward()
                    optimizer.step()


                train_pred=net(train_features[:,:]).reshape(-1)
                test_pred=net(test_features[:,:]).reshape(-1)

                # print(test_pred.shape,test_lebels.shape)
                train_rmse=RMSE(train_pred,train_labels)
                test_rmse=RMSE(test_pred,test_lebels)
                train_mae=MAE(train_pred,train_labels)
                test_mae=MAE(test_pred,test_lebels)

                tbar.set_postfix(min_test_loss=min_rmse.item(), train_loss=train_rmse.item(), test_loss=test_rmse.item(), early_stop=f"{cnt}/{early_stop}")
                tbar.update()

                if min_rmse>test_rmse:
                    min_rmse=test_rmse
                else:
                    cnt+=1

                if cnt==early_stop:
                    break


        # print((torch.abs(test_pred-test_lebels).sum())/test_pred.shape[0])
        test_pred=net(test_features[:,:]).reshape(-1)
        train_pred=net(train_features[:,:]).reshape(-1)

        logger.info({"unit":i,"station_name":station_name,"net":net,
                     "test_rmse":MAE(test_pred,test_lebels).item(),
                     "test_mae":MAE(test_pred,test_lebels).item(),
                     "train_rmse":RMSE(train_pred,train_labels).item(),
                     "train_mae":MAE(train_pred,train_labels).item()
                     })



```
